### PR TITLE
fix cluster role aggregation

### PR DESCRIPTION
**Pull request by s-urbaniak**
cc @sttts @ncdc

### CODE REVIEW

File: pkg/genericcontrolplane/clientutils/multiclusterconfig.go [hunk 1]
```diff
@@ -88,11 +88,41 @@ func defaultRequestInfoResolver() genericapirequest.RequestInfoResolver {
...
+			req.URL.Path = path
+			for i := 0; i < 2 && len(req.URL.RawPath) > 1; i++ {
+				if slash := strings.Index(req.URL.RawPath[1:], "/"); slash != -1 {
+					req.URL.RawPath = req.URL.RawPath[slash:]
+				}
+			}
+		}
+	} else {
+		requestClusterName = req.Header.Get("X-Kubernetes-Cluster")
+	}
```
Comment (sttts) [inline pkg/genericcontrolplane/clientutils/multiclusterconfig.go:106]: please add go doc oneliner on-top. This is non-trivial code. We shouldn't force every reader to read 15 lines of string manipulation.

File: pkg/genericcontrolplane/clientutils/multiclusterconfig.go [hunk 2]
```diff
@@ -88,11 +88,41 @@ func defaultRequestInfoResolver() genericapirequest.RequestInfoResolver {
...
+		contextCluster = &genericapirequest.Cluster {
+			Parents: []string{},
+		}
+		if requestClusterName == "*" {
+			contextCluster.Wildcard = true
+		} else {
+			contextCluster.Name = requestClusterName
+		}
+		req = req.WithContext(genericapirequest.WithCluster(req.Context(), *contextCluster))
+	}
```
Comment (sttts) [inline pkg/genericcontrolplane/clientutils/multiclusterconfig.go:120]: same here, go doc please (one-liners, not paragraphs)

File: pkg/genericcontrolplane/clientutils/multiclusterconfig.go (hunk 1 above)
Comment (sttts) [reply pkg/genericcontrolplane/clientutils/multiclusterconfig.go:106]: @ncdc is all of this going away with scopes? Then I don't care about ^ so much.

File: pkg/controller/clusterroleaggregation/clusterroleaggregation_controller.go [hunk 3]
```diff
@@ -140,8 +140,8 @@ func (c *ClusterRoleAggregationController) syncClusterRole(key string) error {
 	return err
 }
 
-func (c *ClusterRoleAggregationController) applyClusterRoles(name string, newPolicyRules []rbacv1.PolicyRule) error {
-	clusterRoleApply := rbacv1ac.ClusterRole(name).
+func (c *ClusterRoleAggregationController) applyClusterRoles(sharedClusterRole *rbacv1.ClusterRole, newPolicyRules []rbacv1.PolicyRule) error {
+	clusterRoleApply := rbacv1ac.ClusterRole(sharedClusterRole.Name).WithClusterName(sharedClusterRole.ClusterName).
```
[the remark this answers is not in the document]
Comment (sttts) [reply pkg/controller/clusterroleaggregation/clusterroleaggregation_controller.go:144]: nvmd. Got it.

File: pkg/genericcontrolplane/clientutils/multiclusterconfig.go (hunk 1 above)
Comment (ncdc) [reply pkg/genericcontrolplane/clientutils/multiclusterconfig.go:106]: FYI it's mostly a copy of what's in kcp's server.go WithClusterScope handler

**Pull request merged**